### ISSUE TITLE
SimpleConstructorResolverStrategy ignores generic parameter types

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#64]
```
What steps will reproduce the problem?
See attached test case...

What is the expected output? What do you see instead?
List elements are treated/mapped as Object instead of their actual type.

Please use labels and text to provide additional information.


```

Original issue reported on code.google.com by `matt.deb...@gmail.com` on 2 Nov 2012 at 12:20

Attachments:
- [Issue64TestCase.java](https://storage.googleapis.com/google-code-attachments/orika/issue-64/comment-0/Issue64TestCase.java)

**Comment by GoogleCodeExporter** [comment]
```
Fix pushed to current master; for inclusion in next release.
```

Original comment by `matt.deb...@gmail.com` on 2 Nov 2012 at 12:39
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.4.0 release
```

Original comment by `matt.deb...@gmail.com` on 19 Dec 2012 at 2:08

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 19 Dec 2012 at 2:08
- Changed state: **Fixed**
